### PR TITLE
Add GitHub token authentication support

### PR DESCRIPTION
**Pull request by parabuzzle**
Adds the Github token authentication method.

Resolves: https://github.com/hashicorp/vault-ruby/issues/33

The integration tests for this feature rely on the currently unreleased version of Vault. Once the changes that fix https://github.com/hashicorp/vault/issues/572 have been released, the tests for this feature will pass.

### CODE REVIEW

File: lib/vault/api/auth.rb [hunk 1]
```diff
@@ -98,5 +98,27 @@ def userpass(username, password, options = {})
       client.token = secret.auth.client_token
       return secret
     end
+
+    # Authenticate via the Github authentication method. If authentication is
```
Comment (sethvargo) [inline lib/vault/api/auth.rb]: The "H" is capital in GitHub :smile:

Comment (parabuzzle) [inline lib/vault/api/auth.rb]: so very true :smile:  I'll fix that later today.

**Comment by parabuzzle** [comment]
is this just not going to be supported or is there another plan for this?

**Comment by sethvargo** [comment]
d40a10611d9da7e60f7aa536717e0b5c2f1bf05f

**Comment by parabuzzle** [comment]
ah.. ok.. nevermind then... I'm an idiot. thanks seth :)

**Pull request closed without merging**